File: src/test/java/com/github/ccguyka/showupdates/objects/ProjectUpdatesSerializationTest.java
package com.github.ccguyka.showupdates.objects;

import static com.google.common.collect.Lists.newArrayList;
import static org.assertj.core.api.Assertions.assertThat;
import java.io.IOException;

import org.junit.Before;
import org.junit.Test;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.SerializationFeature;
import com.google.common.base.Charsets;
import com.google.common.io.Resources;

public class ProjectUpdatesSerializationTest {

    private ObjectMapper objectMapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);

    private String mavenUpdates;
    private ProjectUpdates updates;

    @Before
    public void setUp() throws IOException {
        mavenUpdates = Resources.toString(this.getClass().getResource("maven-updates.json"), Charsets.UTF_8);
        DependencyUpdates update = new DependencyUpdates(newArrayList(new ArtifactUpdate("test:test", "1", newArrayList("2"))));
        updates = ProjectUpdates.builder()
                .withParent(update)
                .withDependency(update)
                .withDependencyManagement(update)
                .withPlugin(update)
                .build();
    }

    @Test
    public void serialize() throws Exception {

        String json = objectMapper.writeValueAsString(updates);

        assertThat(json).isEqualTo(mavenUpdates);
    }

    @Test
    public void deserialize() throws Exception {

        ProjectUpdates readUpdates = objectMapper.readValue(mavenUpdates, ProjectUpdates.class);

        assertThat(readUpdates).isEqualTo(updates);
    }
}

File: src/test/kotlin/com/github/ccguyka/showupdates/objects/ProjectUpdatesSerializationTest.kt
package com.github.ccguyka.showupdates.objects

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import com.github.ccguyka.showupdates.objects.ProjectUpdates
import com.github.ccguyka.showupdates.objects.ProjectUpdates.Companion.builder
import com.google.common.base.Charsets
import com.google.common.collect.Lists
import com.google.common.io.Resources
import org.assertj.core.api.Assertions
import org.junit.Before
import org.junit.Test
import java.io.IOException

class ProjectUpdatesSerializationTest {

    private val objectMapper = ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT)
    private var mavenUpdates: String? = null
    private var updates: ProjectUpdates? = null

    @Before
    @Throws(IOException::class)
    fun setUp() {
        mavenUpdates = Resources.toString(this.javaClass.getResource("maven-updates.json"), Charsets.UTF_8)
        val update = DependencyUpdates(Lists.newArrayList(ArtifactUpdate("test:test", "1", Lists.newArrayList("2"))))
        updates = builder()
                .withParent(update)
                .withDependency(update)
                .withDependencyManagement(update)
                .withPlugin(update)
                .build()
    }

    @Test
    @Throws(Exception::class)
    fun serialize() {
        val json = objectMapper.writeValueAsString(updates)
        Assertions.assertThat(json).isEqualTo(mavenUpdates)
    }

    @Test
    @Throws(Exception::class)
    fun deserialize() {
        val readUpdates = objectMapper.readValue(mavenUpdates, ProjectUpdates::class.java)
        Assertions.assertThat(readUpdates).isEqualTo(updates)
    }
}
